libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.kt

```diff
@@ -1399,6 +1399,21 @@ open class OAuthWebviewHelper : KeyChainAliasCallback {
         }
     }
 
+    /**
+     * Automatically log in using the provided UI Bridge API parameters.
+     * @param frontdoorBridgeUrl The UI Bridge API front door bridge API
+     * @param pkceCodeVerifier The PKCE code verifier
+     */
+    internal fun loginWithFrontdoorBridgeUrl(
```

Comment:
@wmathurin - I made this `internal`.  Was there any reason it wasn't before?

Reply:
The packing of front door bridge URL and PKCE in a QR code will be up to the apex or java page.
Hopefully it will use the attributes names we recommend, if it does not, an app could parse them out itself and call this method directly?
I guess if we make the method internal, it will force the server page to use the attributes names in the JSON that we expect.

Reply:
That's very helpful, since I was also starting to wonder where a custom APEX with different structure would tie in.  I'll make this public again!